fixtures/fiber-triangle/index.html

```diff
@@ -1,224 +1,241 @@
+      <code>npm run build</code> or
+      <code>yarn build</code>.
+    </p>
+  </div>
+  <script src="../../build/dist/react.development.js"></script>
+  <script src="../../build/dist/react-dom.development.js"></script>
+  <script src="https://unpkg.com/babel-standalone@6/babel.js"></script>
+  <script type="text/babel">
+  const dotStyle = {
+    position: 'absolute',
+    background: '#61dafb',
+    font: 'normal 15px sans-serif',
+    textAlign: 'center',
+    cursor: 'pointer',
+  };
 
-      class Dot extends React.Component {
-        constructor() {
-          super();
-          this.state = { hover: false };
-        }
-        enter() {
-          this.setState({
-            hover: true
-          });
-        }
-        leave() {
-          this.setState({
-            hover: false
-          });
-        }
-        render() {
-          var props = this.props;
-          var s = props.size * 1.3;
-          var style = {
-            ...dotStyle,
-            width: s + 'px',
-            height: s + 'px',
-            left: (props.x) + 'px',
-            top: (props.y) + 'px',
-            borderRadius: (s / 2) + 'px',
-            lineHeight: (s) + 'px',
-            background: this.state.hover ? '#ff0' : dotStyle.background
-          };
-          return (
-            <div style={style} onMouseEnter={() => this.enter()} onMouseLeave={() => this.leave()}>
-              {this.state.hover ? '*' + props.text + '*' : props.text}
-            </div>
-          );
-        }
-      }
+  const containerStyle = {
+    position: 'absolute',
+    transformOrigin: '0 0',
+    left: '50%',
+    top: '50%',
+    width: '10px',
+    height: '10px',
+    background: '#eee',
+  };
 
-      class SierpinskiTriangle extends React.Component {
-        shouldComponentUpdate(nextProps) {
-          var o = this.props;
-          var n = nextProps;
-          return !(
-            o.x === n.x &&
-            o.y === n.y &&
-            o.s === n.s &&
-            o.children === n.children
-          );
-        }
-        render() {
-          let {x, y, s, children} = this.props;
-          if (s <= targetSize) {
-            return (
-              <Dot
-                x={x - (targetSize / 2)}
-                y={y - (targetSize / 2)}
-                size={targetSize}
-                text={children}
-              />
-            );
-            return r;
-          }
-          var newSize = s / 2;
-          var slowDown = true;
-          if (slowDown) {
-            var e = performance.now() + 0.8;
-            while (performance.now() < e) {
-              // Artificially long execution time.
-            }
-          }
+  const targetSize = 25;
 
-          s /= 2;
+  class Dot extends React.Component {
+    constructor() {
+      super();
+      this.state = {hover: false};
+      this.enter = this.enter.bind(this);
+      this.leave = this.leave.bind(this);
```

Comment:
These statements might confuse, because override prototype methods in a constructor and those prototype's methods are not used at all. As `babel-plugin-transform-class-properties` is enabled then you can easily declare those methods using arrow functions:
```
class Dot extends React.Component {
  constructor() {...}
...
  enter = () => {
    this.setState({hover: true})
  }
...
}
```

Reply:
Thanks for your advice 😄 
But I just want to use the **Standard JS** feature in the pure HTML file. This is also consistent with the (doc)(https://reactjs.org/docs/handling-events.html).



> because override prototype methods in a constructor and those prototype's methods are not used at all

The syntax doesn't override prototype methods. For `this.enter  = this.enter.bind(this)`,
the `this.enter` on the left, is a function in the Class Instance, And the `this.enter` on the right, is a function in the prototype.